src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -489,25 +508,29 @@ public Germplasm getByGIDWithPrefAbbrev(final Integer gid) {
 	}
 
 	/**
-	 * Only returns IDs of germplasm with progeny (Germplasm Id is assigned to other germplasm's gpid1, gpid2 OR progntrs.pid)
+	 * Only returns IDs of germplasm with progeny (Germplasm Id is assigned to other germplasm's gpid1, gpid2 OR progntrs.pid) and it's specified method type
 	 *
 	 * @param gids - gids to be checked for descendants
-	 * @return
+	 * @param mtypes - Optional set of method type codes
+	 * @return Set of Integer
 	 */
-	public Set<Integer> getGidsOfGermplasmWithDescendants(final Set<Integer> gids) {
+	public Set<Integer> getGidsOfGermplasmWithDescendants(final Set<Integer> gids, final Set<String> mtypes) {
 		try {
-			if (!CollectionUtils.isEmpty(gids)) {
-				final SQLQuery query = this.getSession().createSQLQuery("SELECT g.gid as gid FROM germplsm g "
-					+ "WHERE (EXISTS (SELECT 1 FROM germplsm descendant WHERE g.gid = descendant.gpid1 OR g.gid = descendant.gpid2 ) "
-					+ "OR EXISTS (SELECT 1 FROM progntrs p WHERE  g.gid = p.gid)) "
-					+ "AND g.gid IN (:gids) AND  g.deleted = 0 AND g.grplce = 0 ");
+			if (!CollectionUtils.isEmpty(gids) && CollectionUtils.isEmpty(mtypes)) {
+				final SQLQuery query = this.getSession().createSQLQuery(FIND_GERMPLASM_WITHDESCENDANTS);
+				query.addScalar("gid", new IntegerType());
+				query.setParameterList("gids", gids);
+				return Sets.newHashSet(query.list());
+			} else if (!CollectionUtils.isEmpty(gids) && !CollectionUtils.isEmpty(mtypes)) {
```

Comment:
I think it's better to create a new method to get the progenies specifically for DER/MAN. Instead of having an optional parameter here. A separately named method reveals the intention better. You can name the new method as "getGidsOfGermplasmWithDerivativeOrMaintenanceDescendants".